src/relax/transform/fuse_ops.cc

```diff
@@ -1238,10 +1238,14 @@ class CompositeFunctionAnnotator : public ExprMutator {
 
   Expr VisitExpr_(const FunctionNode* func_node) final {
     Function f_inner = Downcast<Function>(ExprMutator::VisitExpr_(func_node));
-    auto composite_name = func_node->GetAttr<String>(attr::kComposite);
+
+    if (!func_node->GetAttr<String>(attr::kComposite)) {
```

Comment:
Are non-composite functions visited? https://github.com/apache/tvm/blob/76c1708018d2bdcf4d5a207100ff8682be6cf09d/src/relax/transform/fuse_ops.cc#L1224 here we only visit composite functions

Reply:
The `PatternBasedPartitioner` only visits non-composite functions, produces a composite function for each pattern match, and updates the non-composite function to call the newly-generated composite function.  Afterwards, the call to [`CompositeFunctionAnnotator`](https://github.com/apache/tvm/blob/main/src/relax/transform/fuse_ops.cc#L1289) is called.  This visits only non-composite functions, finds any relax-to-relax function calls, and asserts that the callee is composite.

The callee will be composite for every function call generated by `PatternBasedPartitioner`, but that doesn't guarantee that all relax-to-relax function calls have a composite callee.  If the `IRModule` contains a relax-to-relax call prior to `PatternBasedPartitioner`, that callee may be non-composite.  This IRModule would be entirely legal, but would trigger the assert in `CompositeFunctionAnnotator`.

Reply:
So, the problem isn't with calls to inner functions as on line 1224, but with calls to other functions within the IRModule.

Reply:
I think the problem is if the callee is not a global var, the callee function will still be visited, so the fix makes sense to me

Reply:
Whoops, you're right on that one.  It's if there is a inner function in the input `IRModule`. (Apologies, trying to track too many PRs at one time.)